examples/poller.py

```diff
@@ -98,20 +99,24 @@ def updateDB(self, busName: str, msg: str):
             except json.JSONDecodeError as e:
                 print(f"apps.numgen.updateDB(): {e!r}")
             else:
-                orgDbName = self.dbName
-                self.dbs = {"": ""}
-                self.viewerFrame.dbBox.clear()
-                self.viewerFrame.dbBox.addItem("")
+                originalDBs = set(self.dbs)
+                newDBs = set([""])
                 for db in msg.get("db", ()):
                     if all(key in db for key in ("name", "path")):
                         name, path = db["name"], db["path"]
-                        self.dbs[name] = path
-                        self.viewerFrame.dbBox.addItem(name)
+                        newDBs.add(name)
+                        if name not in self.dbs:
+                            self.dbs[name] = path
+                            self.viewerFrame.dbBox.addItem(name)
```

Comment:
This assumes that the new item is always added at the end, which is actually true for now, but it would be better to specify that assumption somewhere, e.g., in the docstring of this method.

Reply:
Okay! As you said, I assumed that the new database is added at the end and it is so weird.
In addition, it is inefficient to send the whole list from `dbmgr` whenever the database list updates.

However, it is just a toy project, I still remain it.